ffi/dart/tool/build_libs.dart

```diff
@@ -39,96 +41,41 @@ Future<void> main(List<String> args) async {
     print(argParser.usage);
     exit(1);
   }
-  final buildStatic = parsed.option(compileTypeKey)! == 'static';
-  final simulator = parsed.flag(simulatorKey);
-  final targetOS = OS.values.firstWhere((o) => o.name == parsed.option(osKey)!);
-  final targetArchitecture = Architecture.values.firstWhere(
-    (o) => o.name == parsed.option(architectureKey)!,
-  );
-
-  final out = Uri.file(parsed.option(fileKey)!);
 
-  final cargoFeatures = parsed.multiOption(cargoFeaturesKey);
-
-  // We assume that the first folder to contain a cargo.toml above the
-  // output directory is the directory containing the ICU4X code.
-  Directory icu4xPath = recurseToParentRustCrate(File.fromUri(Platform.script));
-
-  final libFileName = targetOS.filename(buildStatic)(
-    crateName.replaceAll('-', '_'),
-  );
-  await buildLib(
-    targetOS,
-    targetArchitecture,
-    buildStatic,
-    simulator,
-    out,
-    libFileName,
-    icu4xPath.path,
-    cargoFeatures,
+  final lib = await buildLib(
+    OS.values.firstWhere((o) => o.name == parsed.option(osKey)!),
+    Architecture.values.firstWhere(
+      (o) => o.name == parsed.option(architectureKey)!,
+    ),
+    parsed.option(compileTypeKey)! == 'static',
+    parsed.flag(simulatorKey),
+    File.fromUri(Platform.script).parent,
+    parsed.multiOption(cargoFeaturesKey),
   );
-}
-
-Directory recurseToParentRustCrate(FileSystemEntity startingPoint) {
-  var folder =
-      startingPoint is Directory ? startingPoint : startingPoint.parent;
-  while (!File.fromUri(folder.uri.resolve('Cargo.toml')).existsSync()) {
-    folder = folder.parent;
-    if (folder.parent == folder) {
-      throw ArgumentError(
-        'Running in the wrong directory, as no Cargo.toml exists above $startingPoint',
-      );
-    }
-  }
-  return folder;
-}
-
-Future<Uri> buildLibraryFromInput(
```

Comment:
Why was this deleted? I do plan on using this once the package is published (which ideally would happen now). It should probably be moved to a `lib/` folder to be part of the public API instead.

Reply:
I was careful to keep the public API and CLI the same :)

if you want this package to be published, then build and links hook should happen in this package, so it wouldn't need to be public.

I'm still not convinced this should be published before native assets stabilise, it's already broken on 3.8.0-beta again.